### PR TITLE
Utility to create taz geofences out of shapefiles

### PR DESCRIPTION
**Pull request by dimaopen**
1. Utility to create taz geofences out of shapefiles
2. Unit test for RideHail TAZ geofence

- - -
This change is [Reviewable](https://reviewable.io/reviews/LBNL-UCB-STI/beam/3716)

### CODE REVIEW

**Comment by dimaopen** [comment]
Test!

File: src/main/scala/scripts/shapefiles/PolygonShapesToTazIdsMapping.scala [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+import org.opengis.referencing.operation.MathTransform
+
+import java.io.File
+import java.nio.file.{Path, Paths}
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Do mapping of taz ids to the polygons in the shape file.
+  * If the TAZ center is within a polygon then that TAZ belongs to that polygon.
+  * Each set of tas ids is saved as a separate csv file (named {feature-id}.csv) with a single column (taz).
```
Comment (nikolayilyin) [inline src/main/scala/scripts/shapefiles/PolygonShapesToTazIdsMapping.scala]: a typo: 'Each set of tas ids' needs to be fixed as 'Each set of TAZ ids'

File: src/main/scala/scripts/shapefiles/PolygonShapesToTazIdsMapping.scala [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  )
+
+  private def parseArgs(args: Array[String]): Option[CliOptions] = {
+    import scopt.OParser
+    val builder = OParser.builder[CliOptions]
+    val parser1 = {
+      import builder._
+      OParser.sequence(
+        programName("shp-to-csv-converter"),
+        head("This program converts an shp file to a csv that contains centroids fo polygons with their ids"),
```
Comment (nikolayilyin) [inline src/main/scala/scripts/shapefiles/PolygonShapesToTazIdsMapping.scala]: another typo: 'converts an shp' needs to be fixed as 'converts a shp'

**Review by nikolayilyin — APPROVED** [review]
A couple of small typos, the rest looks good to me

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test now!

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test now!

**Comment by dimaopen** [comment]
Test!

**Pull request merged**